File: DrawDemo1/app/src/main/java/edu/cs4730/drawdemo1/MainActivity.java
package edu.cs4730.drawdemo1;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import edu.cs4730.drawdemo1.databinding.ActivityMainBinding;

/**
 * while there is alot of code here, the example is actually all in the fragments.
 * this is just to get the bottom navigation bar  work.
 */

public class MainActivity extends AppCompatActivity {
    ActivityMainBinding binding;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        setSupportActionBar(binding.toolbar);
        
        //setup a listener, which acts very similar to how menus are handled.
        binding.navView.setOnItemSelectedListener(new BottomNavigationView.OnItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                int id = menuItem.getItemId();
                if (id == R.id.drawing) {
                    getSupportFragmentManager().beginTransaction().replace(binding.container.getId(), new MainFragment()).commit();
                    menuItem.setChecked(true);  //make sure to check/highlight the item.
                    return true;
                } else if (id == R.id.drawing1) {
                    menuItem.setChecked(true); //make sure the item is checked/highlighted
                    getSupportFragmentManager().beginTransaction().replace(binding.container.getId(), new Draw1Fragment()).commit();
                    return true;
                } else if (id == R.id.andraw) {
                    getSupportFragmentManager().beginTransaction().replace(binding.container.getId(), new AnDrawFragment()).commit();
                    menuItem.setChecked(true); //make sure the item is checked/highlighted
                    return true;
                } else if (id == R.id.asdraw) {

                    menuItem.setChecked(true); //make sure the item is checked/highlighted
                    getSupportFragmentManager().beginTransaction().replace(binding.container.getId(), new AsycDrawFragment()).commit();
                    return true;
                }
                return false;
            }
        });

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .add(binding.container.getId(), new MainFragment()).commit();
        }
    }
}

File: DrawDemo1_kt/app/src/main/java/edu/cs4730/drawdemo1_kt/MainActivity.kt
package edu.cs4730.drawdemo1_kt

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.google.android.material.navigation.NavigationBarView
import edu.cs4730.drawdemo1_kt.databinding.ActivityMainBinding

/**
 * while there is alot of code here, the example is actually all in the fragments.
 * this is just to get the bottom navigation bar  work.
 */

class MainActivity : AppCompatActivity() {
    lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(binding.main) { v: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(
                systemBars.left, systemBars.top, systemBars.right, systemBars.bottom
            )
            WindowInsetsCompat.CONSUMED
        }
        setSupportActionBar(binding.toolbar)

        //setup a listener, which acts very similar to how menus are handled.
        binding.navView.setOnItemSelectedListener(NavigationBarView.OnItemSelectedListener { menuItem ->
            val id = menuItem.itemId
            if (id == R.id.drawing) {
                supportFragmentManager.beginTransaction()
                    .replace(binding.container.id, MainFragment()).commit()
                menuItem.isChecked = true //make sure to check/highlight the item.
                return@OnItemSelectedListener true
            } else if (id == R.id.drawing1) {
                menuItem.isChecked = true //make sure the item is checked/highlighted
                supportFragmentManager.beginTransaction()
                    .replace(binding.container.id, Draw1Fragment()).commit()
                return@OnItemSelectedListener true
            } else if (id == R.id.andraw) {
                supportFragmentManager.beginTransaction()
                    .replace(binding.container.id, AnDrawFragment()).commit()
                menuItem.isChecked = true //make sure the item is checked/highlighted
                return@OnItemSelectedListener true
            } else if (id == R.id.asdraw) {
                menuItem.isChecked = true //make sure the item is checked/highlighted
                supportFragmentManager.beginTransaction()
                    .replace(binding.container.id, AsycDrawFragment()).commit()
                return@OnItemSelectedListener true
            }
            false
        })
        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction().add(binding.container.id, MainFragment())
                .commit()
        }
    }
}